packages/manager/src/features/Databases/DatabaseDetail/AddAccessControlDrawer.tsx

```diff
@@ -68,9 +68,17 @@ const AddAccessControlDrawer: React.FC<CombinedProps> = (props) => {
     }
   ) => {
     // Get the IP address strings out of the objects and filter empty strings out.
+    // Ensure we append /32 to all IPs if not already present.
     const allowListRetracted = _allowList
```

Comment:
Since the chained array methods are starting to pile up now (3), we can consider using `.reduce()` instead. We'll avoid looping over the array 3 separate times, while still getting the IP strings, filtering out the empty strings, and appending `/32` if needed.

```suggestion
    const allowListRetracted = _allowList.reduce((acc, currentIP) => {
      let ipString = extendedIPToString(currentIP);
      if (ipString === '') {
        return acc;
      }

      if (ipString.indexOf('/32') === -1) {
        ipString += '/32';
      }

      return [...acc, ipString];
    }, []);
```